WordPress/WordPressTest/PostRepositoryTests.swift

```diff
@@ -0,0 +1,140 @@
+import XCTest
+
+@testable import WordPress
+
+class PostRepositoryTests: CoreDataTestCase {
+
+    private var remoteMock: PostServiceRESTMock!
+    private var repository: PostRepository!
+    private var blogID: CoreDataObjectIdentifier<Blog>!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+
+        let accountService = AccountService(coreDataStack: contextManager)
+        let accountID = accountService.createOrUpdateAccount(withUsername: "username", authToken: "token")
+        try accountService.setDefaultWordPressComAccount(XCTUnwrap(mainContext.existingObject(with: accountID) as? WPAccount))
+
+        let blog = try BlogBuilder(mainContext).withAccount(id: accountID).build()
+
+        contextManager.saveContextAndWait(mainContext)
+
+        blogID = .ofSaved(blog)
+        remoteMock = PostServiceRESTMock()
+        let remoteFactory = PostServiceRemoteFactoryMock()
+        remoteFactory.remoteToReturn = remoteMock
+        repository = PostRepository(coreDataStack: contextManager, remoteFactory: remoteFactory)
+    }
+
+    func testGetPost() async throws {
+        let post = RemotePost(siteID: 1, status: "publish", title: "Post: Test", content: "This is a test post")
+        post?.type = "post"
+        remoteMock.remotePostToReturnOnGetPostWithID = post
+        let postID = try await repository.getPost(withID: 1, from: blogID)
+        let isPage = try await contextManager.performQuery { try postID.existingObject(in: $0) is Page }
+        let title = try await contextManager.performQuery { try postID.existingObject(in: $0).postTitle }
+        let content = try await contextManager.performQuery { try postID.existingObject(in: $0).content }
+        XCTAssertFalse(isPage)
+        XCTAssertEqual(title, "Post: Test")
+        XCTAssertEqual(content, "This is a test post")
+    }
+
+    func testGetPage() async throws {
+        let post = RemotePost(siteID: 1, status: "publish", title: "Post: Test", content: "This is a test post")
+        post?.type = "page"
+        remoteMock.remotePostToReturnOnGetPostWithID = post
+        let postID = try await repository.getPost(withID: 1, from: blogID)
+        let isPage = try await contextManager.performQuery { try postID.existingObject(in: $0) is Page }
+        let title = try await contextManager.performQuery { try postID.existingObject(in: $0).postTitle }
+        let content = try await contextManager.performQuery { try postID.existingObject(in: $0).content }
+        XCTAssertTrue(isPage)
+        XCTAssertEqual(title, "Post: Test")
+        XCTAssertEqual(content, "This is a test post")
+    }
+
+}
+
+// These mock classes are copied from PostServiceWPComTests. We can't simply remove the `private` in the original class
+// definition, because Xcode would complian about 'WordPress' module not found.
```

Comment:
This is sad 😞 

From what I can understand, making the type `internal` adds it to the Objective-C generated header `WordPressTest-Swift.h`, together with a whole bunch of additional imports that weren't there before. 

I expected that header to be generated after the WordPress target has been compiled and therefor for the module to be available. (The WordPress target has "Defines Module: Yes")

Maybe that actually happens, and the issue has more to do with search paths?

Regardless, that's something we'll have to address at another time 👌